lxd/cluster/gateway.go

```diff
@@ -689,7 +689,11 @@ func (g *Gateway) LeaderAddress() (string, error) {
 	transport, cleanup := tlsTransport(config)
 	defer cleanup()
 
+	var ctx context.Context
```

Comment:
This change looks a little odd. I can understand why the linter was complaining about it.
But the fix looks quite confusing and has a problem that the old version didn't: the return on error on line 706 wont cancel the context.

How about making each client attempt its own function which can then use defer to cancel the context?

Reply:
I've moved this into a separate function.